plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -648,13 +648,19 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             const auto block_deadline = calculate_block_deadline( chain.pending_block_time() );
             push_result pr = push_transaction( block_deadline, trx, persist_until_expired, return_failure_trace, next );
 
-            exhausted = pr.block_exhausted;
+            bool exhausted = pr.block_exhausted;
             if( pr.trx_exhausted ) {
                _unapplied_transactions.add_incoming( trx, persist_until_expired, return_failure_trace, next );
             } else if( pr.persist ) {
                _unapplied_transactions.add_persisted( trx );
             }
 
+            success = !exhausted; // transaction failed if block was exhausted
+
+            if (!(_pending_block_mode == pending_block_mode::producing) && pr.trx_exhausted)
```

Comment:
Is there a reason to write `!(_pending_block_mode == pending_block_mode::producing)` vs `_pending_block_mode != pending_block_mode::producing` or even `_pending_block_mode == pending_block_mode::speculating` ?

Reply:
I think this read the most as `if not in cprducing mode`, but I have added members to make it clearer.